app/src/main/java/com/hover/stax/transfers/TransferFragment.java

```diff
@@ -170,8 +170,18 @@ private void setRecipientHint(HoverAction action) {
 		summaryCard.findViewById(R.id.recipientRow).setVisibility(action.requiresRecipient() ? View.VISIBLE : View.GONE);
 		if (!action.requiresRecipient())
 			recipientValue.setContent(getString(R.string.self_choice), "");
-		else
+		else {
+			addRecipientsValue();
 			contactInput.setHint(action.getRequiredParams().contains(HoverAction.ACCOUNT_KEY) ? getString(R.string.recipientacct_label) : getString(R.string.recipientphone_label));
+		}
+	}
+
+	private void addRecipientsValue() {
+		StaxContact contact = transferViewModel.getContact().getValue();
```

Comment:
Ideal usage of ViewModels is to let the observer handle the data change and update the UI. We shouldn't fetch the value directly from the MutableLiveData object.

Reply:
Very true, thanks for this correction.